### DISCUSSION TITLE
change constant primer problem

### DISCUSSION THREAD

**Discussion by bigcat1001** [milaboratory/mixcr#1360] [Q&A] [answered]
Hi,
     I used TAKARA method to construct library and the "takara-human-bcr-cdr3" preset works well.However,when I changed the constant IgG primer to the 3' direction,the mixcr alignment become bad.
     `Analysis date: Tue Sep 19 12:56:45 CST 2023
Input file(s): SQ23040119-0906-2-0905-4_combined_R1.fastq.gz,SQ23040119-0906-2-0905-4_combined_R2.fastq.gz
Output file(s): ./Sample_SQ23040119-0906-2-0905-4_B_output-cdr3.vdjca
Version: 4.1.2; built=Mon Nov 28 17:16:06 CST 2022; rev=ec9c6fd320; lib=repseqio.v2.0
Command line arguments: align --report ./Sample_SQ23040119-0906-2-0905-4_B_output-cdr3.align.report.txt --json-report ./Sample_SQ23040119-0906-2-0905-4_B_output-cdr3.align.report.json --threads 40 --preset takara-human-bcr-cdr3 --tag-pattern ^N{8}(R1:*)\^UMI:NNNNTtNNNNTNNNNT(R2:*) SQ23040119-0906-2-0905-4_combined_R1.fastq.gz SQ23040119-0906-2-0905-4_combined_R2.fastq.gz ./Sample_SQ23040119-0906-2-0905-4_B_output-cdr3.vdjca
Analysis time: 1.65m
Total sequencing reads: 1079468
Successfully aligned reads: 28232 (2.62%)
Paired-end alignment conflicts eliminated: 116 (0.01%)
Alignment failed, no hits (not TCR/IG?): 242062 (22.42%)
Alignment failed because of absence of V hits: 6156 (0.57%)
Alignment failed because of absence of J hits: 23494 (2.18%)
No target with both V and J alignments: 557254 (51.62%)
Absent barcode: 222270 (20.59%)
Overlapped: 234843 (21.76%)
Overlapped and aligned: 28066 (2.6%)
Alignment-aided overlaps: 24 (0.09%)
Overlapped and not aligned: 206777 (19.16%)
No CDR3 parts alignments, percent of successfully aligned: 2 (0.01%)
Partial aligned reads, percent of successfully aligned: 21 (0.07%)
V gene chimeras: 6 (0%)
J gene chimeras: 1 (0%)
TRA chains: 2 (0.01%)
TRA non-functional: 0 (0%)
TRB chains: 16 (0.06%)
TRB non-functional: 0 (0%)
IGH chains: 28213 (99.93%)
IGH non-functional: 203 (0.72%)
IGK chains: 1 (0%)
IGK non-functional: 0 (0%)
Realigned with forced non-floating bound: 1244758 (115.31%)
Realigned with forced non-floating right bound in left read: 1558 (0.14%)
Realigned with forced non-floating left bound in right read: 1558 (0.14%)
Tag parsing report:
  Execution time: 0ns
  Total reads: 1079468
  Matched reads: 857198 (79.41%)
  Projection +R1 +R2: 857198 (79.41%)
  For variant 0:
    For projection IntListKey(list=[1, 2]):
      R1:Left position: 8
      R1:Right position: 150
      UMI:Left position: 5
      R2:Left position: 21
      UMI:Right position: 21
      Variants: 0
      Cost:
        0: + 842923 (98.33%) = 842923 (98.33%)
        2: + 14275 (1.67%) = 857198 (100%)
      R1 length: 142
      UMI length: 16
      R2 length: 129`

It seems only 2% reads matched.But I manual checked fastq file and blast some reads.It performs well.About 90% reads are BCR reads.I assumed the constant primer should be changed in preset? How can I do it?
My prior constant primer is 'AAGACCGATGGGCCCTTG' and my new primer is 'GAAGTAGTCCTTGACCAGGC'

**Comment by mizraelson** [comment, marked as answer]
Hi,

First, I strongly advise updating to the most recent version of MiXCR, as there have been updates tailored to this preset. Could you also provide the exact command you used?

By modifying the primer location, you've shifted 73bp further from the CDR3 region, which I suspect makes it challenging to entirely encompass the `CDR3` region with a 150bp read.

Here's an elucidation of the more salient report fields for your sample:

- `Alignment failed, no hits (not TCR/IG?): 242062 (22.42%)`: These are non-target reads that didn't align to any TCR or BCR reference. To manually inspect these reads, you can utilize the `--output-not-used-reads` parameter with `mixcr analyze`.
  
- `No target with both V and J alignments: 557254 (51.62%)`: This indicates that for roughly 50% of reads, one read covers only the J gene while the other encompasses only the V gene. As a result, there's likely a gap within the CDR3 region. This phenomenon likely stems from relocating the primer without extending the read length.
  
- `Absent barcode: 222270 (20.59%)`: This result is associated with the UMI pattern and shouldn't appear in the latest MiXCR version if you're employing the standard Takara kit UMI. If you're utilizing a custom UMI, I'd suggest using `(UMI:N{16})`. Bear in mind that a standard Takara UMI consists of 12bp.
  
- `Overlapped: 234843 (21.76%)`: This outcome is concerning. Given the parameters of 5'RACE and a 150+150 read length, overlaps shouldn't occur at all. This might signify the presence of shorter, non-specific sequences that weren't adequately eliminated during the wet-lab library preparation. It's plausible these sequences correspond to the off-target reads from the first report field.

If possible, share some sample reads for both R1 and R2 (around 15 pairs), so I can provide a more detailed analysis and visualize the potential issues.

Sincerely,
Mark
